src/uterm_input_uxkb.c

```diff
@@ -158,6 +163,47 @@ int uxkb_desc_init(struct uterm_input *input,
 			   model, layout, variant, options);
 	}
 
+	locale = getenv("LC_ALL");
+	if (!locale || !*locale) {
+		locale = getenv("LC_CTYPE");
+	}
+	if (!locale || !*locale) {
+		locale = getenv("LANG");
+	}
+	if (!locale || !*locale) {
+		locale = "C";
+	}
+
+	if (compose_file && *compose_file) {
+		compose_filp = fopen(compose_file, "rb");
+		if (compose_filp) {
+			input->compose_table = xkb_compose_table_new_from_file(
+						input->ctx,
+						compose_filp,
+						locale,
+						XKB_COMPOSE_FORMAT_TEXT_V1,
+						0);
+
+			if (!input->compose_table) {
+				llog_warn(input, "cannot load compose file, "
+					  "reverting to system default");
+			}
+
+			fclose(compose_filp);
+		} else {
+			llog_warn(input, "cannot open compose file, "
+				  "reverting to system default");
+		}
+
+	}
+
+	if (!input->compose_table) {
+		input->compose_table = xkb_compose_table_new_from_locale(
+								input->ctx,
+								locale,
+								0);
```

Comment:
Add a warning log here if no compose_table is found? To at least give the user a clue about what's happening.